### PR TITLE
RDB Loader 5.0.0

### PR DESCRIPTION
**Pull request by spenes**
In the RDB Loader 5.0.0, GCP support is brought to Snowflake Loader and new `transformer-pubsub` application is created. I've made necessary changes in the documentation for these new features.

The biggest change I've made is on the transformer config page. Previously, there was just one page for all transformer applications. However,  I realized that if I tried to add configuration for `transformer-pubsub` to that page, it would be really complicated. Therefore, I created individual docs page for all transformer application. However, they have still too many common config fields and repeating them on separate pages would be really hard to maintain later on. Therefore, I extracted out common parts to separate files and combined them in the respective configuration page later on.

I didn't use the same technique in the loader config page because the changes needed in there weren't much therefore I thought it is okay to use existing format.

### CODE REVIEW

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/rdb-loader-previous-versions/rdb-loader-4-0-x/index.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
+---
+title: "RDB Loader 4.0.x"
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/rdb-loader-previous-versions/rdb-loader-4-0-x/index.md:2]: I notice this page: https://deploy-preview-139--snowplow-docs.netlify.app/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/rdb-loader-previous-versions/

doesn't have the card view on it. e.g. https://raw.githubusercontent.com/snowplow/documentation/1a830685de520cc937814bb911ab85aef8e49212/docs/try-snowplow/recipes/index.md

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/index.md [hunk 2]
```diff
@@ -49,11 +58,13 @@ For loading into **Snowflake**, use the `snowplow-rdb-loader-snowflake` artefact
 
 For loading into **Databricks**, use the `snowplow-rdb-loader-databricks` artefact.
 
+At the moment, all loader applications are can be run with AWS services. The aim is making possible to run them with GCP services too. In version 5.0.0, we integrated GCP services to Snowflake Loader. Databricks Loader on GCP is in our roadmap.
+
```
Comment (peel) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/index.md:62]:
```suggestion
:::note

At the moment, all loader applications are can be run with AWS services. The aim is making possible to run them with GCP services too. In version 5.0.0, we integrated GCP services to Snowflake Loader. Databricks Loader on GCP is in our roadmap.
:::
```

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-kinesis/index.md [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+
+See [here](/docs/pipeline-components-and-applications/iglu/iglu-resolver/index.md) for details on how to prepare the Iglu resolver file.
+
+**NOTE:** All self-describing schemas for events processed by the transformer **must** be hosted on [Iglu Server](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-server/index.md) 0.6.0 or above. [Iglu Central](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-central/index.md) is a registry containing Snowplow-authored schemas. If you want to use them alongside your own, you will need to add it to your resolver file. Keep it mind that it could override your own private schemas if you give it higher priority. For details on this see [here](https://discourse.snowplow.io/t/important-changes-to-iglu-centrals-api-for-schema-lists/5720#how-will-this-affect-my-snowplow-pipeline-3).
```
Comment (peel) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-kinesis/index.md:30]:
```suggestion
:::tip

All self-describing schemas for events processed by the transformer **must** be hosted on [Iglu Server](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-server/index.md) 0.6.0 or above. [Iglu Central](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-central/index.md) is a registry containing Snowplow-authored schemas. If you want to use them alongside your own, you will need to add it to your resolver file. Keep it mind that it could override your own private schemas if you give it higher priority. For details on this see [here](https://discourse.snowplow.io/t/important-changes-to-iglu-centrals-api-for-schema-lists/5720#how-will-this-affect-my-snowplow-pipeline-3).

:::
```

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md [hunk 4]
```diff
@@ -9,14 +9,23 @@ import {versions} from '@site/src/componentVersions';
 import CodeBlock from '@theme/CodeBlock';
 ```
 
-There are two ways to set up the necessary Snowflake resources and run the loader:
+It is possible to run Snowflake Loader on both AWS and GCP.
+
+#### Running on AWS
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md:14]: nit: 
```suggestion
### Running on AWS
```

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md [hunk 5]
```diff
@@ -9,14 +9,23 @@ import {versions} from '@site/src/componentVersions';
...
+
+There are two ways to set up the necessary Snowflake resources and run the loader on AWS:
 
 - using our dedicated open source Terraform modules to create the resources and deploy the loader on EC2
 - creating the resources and running the application manually.
 
 We recommend the first way.
 
-## Using the Terraform modules
+#### Running on GCP
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md:23]:
```suggestion
### Running on GCP
```

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/index.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+
+See [here](/docs/pipeline-components-and-applications/iglu/iglu-resolver/index.md) for details on how to prepare the Iglu resolver file.
+
+**NOTE:** All self-describing schemas for events processed by the transformer **must** be hosted on [Iglu Server](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-server/index.md) 0.6.0 or above. [Iglu Central](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-central/index.md) is a registry containing Snowplow-authored schemas. If you want to use them alongside your own, you will need to add it to your resolver file. Keep it mind that it could override your own private schemas if you give it higher priority. For details on this see [here](https://discourse.snowplow.io/t/important-changes-to-iglu-centrals-api-for-schema-lists/5720#how-will-this-affect-my-snowplow-pipeline-3).
```
Comment (peel) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/index.md:35]:
```suggestion
:::tip
All self-describing schemas for events processed by the transformer **must** be hosted on [Iglu Server](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-server/index.md) 0.6.0 or above. [Iglu Central](/docs/pipeline-components-and-applications/iglu/iglu-repositories/iglu-central/index.md) is a registry containing Snowplow-authored schemas. If you want to use them alongside your own, you will need to add it to your resolver file. Keep it mind that it could override your own private schemas if you give it higher priority. For details on this see [here](https://discourse.snowplow.io/t/important-changes-to-iglu-centrals-api-for-schema-lists/5720#how-will-this-affect-my-snowplow-pipeline-3).
:::
```

**Review by peel — APPROVED** [review]

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md [hunk 7]
```diff
@@ -72,7 +81,11 @@ The loader takes two configuration files:
...
+An example of the minimal required config for the Snowflake loader on AWS can be found [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/aws/snowflake.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/aws/snowflake.config.reference.hocon). 
+
+An example of the minimal required config for the Snowflake loader on GCP can be found [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/gcp/snowflake.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/gcp/snowflake.config.reference.hocon). 
+
+For details about each setting, see the [configuration reference](/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/index.md).
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md:88]: Not part of this PR but a couple of lines lower than this is a `note` that could become a `:::note`

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md [hunk 8]
```diff
@@ -72,7 +81,11 @@ The loader takes two configuration files:
...
+An example of the minimal required config for the Snowflake loader on AWS can be found [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/aws/snowflake.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/aws/snowflake.config.reference.hocon). 
+
+An example of the minimal required config for the Snowflake loader on GCP can be found [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/gcp/snowflake.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/loader/gcp/snowflake.config.reference.hocon).
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/snowflake-loader/index.md:86]: This might be nice as a little table?

| Cloud |        Minimal Configuration        |         Extended Configuration        |
|:-----:|:-----------------------------------:|:-------------------------------------:|
| AWS   | /aws/snowflake.config.minimal.hocon | /aws/snowflake.config.reference.hocon |
| GCP   | /gcp/snowflake.config.minimal.hocon | /gcp/snowflake.config.reference.hocon |

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/telemetry/index.md [hunk 9]
```diff
@@ -0,0 +1,23 @@
+---
+title: "Telemetry"
+date: "2022-10-13"
+sidebar_position: 100
+---
+
+Starting with version 4.0.0 of Transformer Kinesis and version 5.0.0 of Transformer Pubsub, Redshift Loader, Snowflake Loader, Databricks Loader, Snowplow collects heartbeats with some meta-information about the application (schema [here](https://raw.githubusercontent.com/snowplow/iglu-central/master/schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-1)). This is done to help us to improve the product, we need to understand what is popular, so we could focus our development effort in the right place.
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/telemetry/index.md:7]:
```suggestion
Snowplow collects heartbeats with some meta-information about the application (schema [here](https://raw.githubusercontent.com/snowplow/iglu-central/master/schemas/com.snowplowanalytics.oss/oss_context/jsonschema/1-0-1)). This is done to help us to improve the product, we need to understand what is popular, so we could focus our development effort in the right place.

Telemetry was introduced in version 4.0.0 of Transformer Kinesis and version 5.0.0 of Transformer Pubsub, Redshift Loader, Snowflake Loader, Databricks Loader.
```

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/index.md (hunk 6 above)
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/index.md:35]: `:::note`

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/index.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
...
+<span>It's also available as a Docker image on Docker Hub under <code>{`snowplow/transformer-pubsub:${versions.rdbLoader}`}</code></span>
+
+## Configuring `snowplow-transformer-pubsub`
+
+The transformer takes two configuration files:
+
+- a `config.hocon` file with application settings
+- an `iglu_resolver.json` file with the resolver configuration for your [Iglu](https://github.com/snowplow/iglu) schema registry.
+
+An example of the minimal required config for the Transformer Pubsub can be found [here](https://github.com/snowplow/snowplow-rdb-loader/tree/master/config/transformer/gcp/transformer.pubsub.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/tree/master/config/transformer/gcp/transformer.pubsub.config.reference.hocon). For details about each setting, see the [configuration reference](/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/configuration-reference).
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/stream-transformer/transformer-pubsub/index.md:31]: docs links should always point to `index.md`

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/spark-transformer/index.md [hunk 11]
```diff
@@ -155,7 +155,7 @@ The transformer takes two configuration files:
...
 
-An example of the minimal required config for the Spark transformer can be found [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/transformer.batch.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/blob/master/config/transformer.batch.config.reference.hocon). For details about each setting, see the [configuration reference](/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/rdb-transformer-configuration-reference/index.md).
+An example of the minimal required config for the Spark transformer can be found [here](https://github.com/snowplow/snowplow-rdb-loader/tree/master/config/transformer/aws/transformer.batch.config.minimal.hocon) and a more detailed one [here](https://github.com/snowplow/snowplow-rdb-loader/tree/master/config/transformer/aws/transformer.batch.config.reference.hocon). For details about each setting, see the [configuration reference](/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/spark-transformer/configuration-reference/).
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/spark-transformer/index.md:158]: Docs links should always end in `/index.md`

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/reusable/aws-only/_index.mdx [hunk 12]
```diff
@@ -0,0 +1,56 @@
...
+    <td><code>queue.region</code></td>
+    <td>AWS region of the SQS queue or SNS topic. Optional if it can be resolved with <a href="https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/regions/providers/DefaultAwsRegionProviderChain.html">AWS region provider chain</a>.</td>
+</tr>
+<tr>
+    <td><code>formats.*</code></td>
+    <td>Schema-specific format settings.</td>
+</tr>
+<tr>
+    <td><code>formats.transformationType</code></td>
+    <td>Required. Type of transformation, either `shred` or `widerow`. See <a href="/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/#shredded-data">Shredded data</a> and <a href="/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/#wide-row-format">Wide row format</a>.</td>
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/transforming-enriched-data/reusable/aws-only/_index.mdx:35]: Docs links should always end in `/index.md`

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/index.md [hunk 13]
```diff
@@ -13,20 +13,29 @@ Loading [enriched](/docs/pipeline-components-and-applications/enrichment-compone
 
 Each step is handled by a dedicated application: `transformer` or `loader`. To load Snowplow data with RDB Loader, you'll need to run one of each.
 
+**Note:** Snowplow collects heartbeats from RDB Loader applications with some meta-information about the application. More information can be found in [here](/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/telemetry).
```
Comment (paulboocock) [inline docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/index.md:16]: Docs links should always end in `/index.md`

File: docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/index.md [hunk 14]
```diff
@@ -4,15 +4,17 @@ date: "2022-04-05"
 sidebar_position: 500
 ---
 
-The configuration reference in this page is written for RDB Loader 4.0.0 or higher
+The configuration reference in this page is written for RDB Loader 5.0.0 or higher
```
[the remark this answers is not in the document]
Comment (spenes) [reply docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/index.md:7]: There are two types of previous versions folder:
* [The one for previous config references](https://docs.snowplow.io/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/loading-transformed-data/rdb-loader-configuration-reference/rdb-loader-previous-versions/)
* [The one for main rdb loader section](https://docs.snowplow.io/docs/pipeline-components-and-applications/loaders-storage-targets/snowplow-rdb-loader/previous-versions/)

I kept the config reference for v4 however we discussed with @stanch and decided that it would better if we don't create v4 section under main previous versions folder because most of the parts are still same. Maybe we can discuss later to decide structure of previous versions folder and what to put there exactly.
